crates/gs_client/src/states/in_game.rs

```diff
@@ -1,19 +1,97 @@
 //! The state for when the player is in game, with all basic gameplay resources fully loaded.
 
 use bevy::prelude::*;
+use gs_common::prelude::GenericAsyncResult;
+use gs_common::raycast::{RaycastContext, raycast};
+use gs_common::voxel::blocks::STONE_BLOCK_NAME;
+use gs_common::voxel::plugin::BlockRegistryHolder;
+use gs_schemas::actions::{PositionData, ThrowAction};
+use gs_schemas::coordinates::WorldPos;
+use gs_schemas::raycast::{RaycastHitMask, RaycastResult, RaycastSpec};
+use gs_schemas::voxel::chunk_storage::ChunkStorage;
+use gs_schemas::voxel::voxeltypes::{BlockEntry, EMPTY_BLOCK_NAME};
 
 use crate::ClientNetworkThreadHolder;
 use crate::states::ClientAppState;
+use crate::voxel::ClientVoxelUniverse;
 
 /// The "plugin" implementing the in game state.
 pub struct InGamePlugin;
 
 impl Plugin for InGamePlugin {
     fn build(&self, app: &mut App) {
+        app.init_resource::<InGamePromiseHolder>();
         app.add_systems(OnExit(ClientAppState::InGame), ingame_cleanup_on_exit);
     }
 }
 
+#[derive(Resource, Default)]
+pub(crate) struct InGamePromiseHolder {
+    promises: Vec<Box<dyn GenericAsyncResult + Send + Sync>>,
+}
+
 fn ingame_cleanup_on_exit(net_thread: ResMut<ClientNetworkThreadHolder>) {
     net_thread.0.sync_shutdown();
 }
+
+/// sends a throw packet over the given net thread and promise holder
+pub(crate) fn ingame_send_throw_packet(
+    net_thread: &Res<ClientNetworkThreadHolder>,
+    promises: &mut ResMut<InGamePromiseHolder>,
+    voxel_query: &mut Query<&mut ClientVoxelUniverse>,
+    bregistry: &Res<BlockRegistryHolder>,
+    position: PositionData,
+    throw: ThrowAction,
+) {
+    promises
+        .promises
+        .push(Box::new(net_thread.0.schedule_task(async move |state| {
```

Comment:
For now I'd skip the entire promise thing here and just directly call `net_thread.0.schedule_task` and forget the result until we have some system in place for watching network errors and reacting to them